SQL breaking changes to fix Swagger spec validation errors/warnings

jaredmoo opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

AutoRest enabled several new validation rules which exposed some naming issues in SQL swagger specs. I also noticed that several classes are no longer supported.
- Removed unsupported types (e.g. Schema, Table, Column)
- Revised import export  operation and class names
- Removed unnecessary import export operation results APIs

Corresponding Swagger change: https://github.com/Azure/azure-rest-api-specs/pull/1005

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

msftclas commented:
@jaredmoo,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

cormacpayne commented:
This PR should be merged at the same time as https://github.com/Azure/azure-sdk-for-net/pull/2914 and https://github.com/Azure/azure-sdk-for-net/pull/2925

jaredmoo commented:
#2914 and #2925 are now combined into this

jaredmoo commented:
Swagger changes are all merged now.

jaredmoo commented:
btw project.json isn't updated because 1.2.0-preview hasn't been published yet.

cormacpayne commented:
@jaredmoo Hey Jared, two quick questions:

1) Just to be sure, does this spec include _all_ of the changes you originally wanted to make in this PR and the two that you closed?

2) There are a lot of unnecessary commits, would you mind [rebasing](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md#rebasing) to get the number of commits down to 1 or 2?

jaredmoo committed:
Capabilities and security policy features.

Also renamed several types and operations for improved clarify and
consistency.

Additions:

- BlobAuditingPolicy APIs (e.g. Databases.CreateOrUpdateBlobAuditingPolicy)
- ThreatDetectionPolicy APIs (e.g. Databases.CreateOrUpdateThreatDetectionPolicy)
- Databases.ListByServer now supports $expand parameter
- Capabilities APIs (e.g. Capabilities.ListByLocation)

Types renamed:

- ServerFirewallRule -> FirewallRule
- DatabaseEditions -> DatabaseEdition
- ElasticPoolEditions -> ElasticPoolEdition
- ImportRequestParameters -> ImportRequest
- ExportRequestParameters -> ExportRequest
- ImportExportOperationResponse -> ImportExportResponse
- OperationMode -> ImportOperationMode
- TransparentDataEncryptionStates -> TransparentDataEncryptionStatus

Types removed:

- Unused types: UpgradeHint, Schema, Table, Column

Operations renamed:

- Servers.GetByResourceGroup -> Servers.Get
- Servers.CreateOrUpdateFirewallRule -> FirewallRule.CreateOrUpdate, and
similar for Get, List, and Delete
- Databases.Import -> Databases.CreateImportOperation
- Servers.Import -> Databases.Import
- Databases.PauseDataWarehouse -> Databases.Pause
- Databases.ResumeDataWarehouse -> Databases.Resume

Operations removed:

- Removed ImportExport operation results APIs since these are handled
  automatically by Azure async pattern.

jaredmoo commented:
Yes, since all swagger changes are now merged, all generated files in this change are based on rest-api-specs master branch.

I have squashed into a single commit and described all the breaking changes.

jaredmoo commented:
@cormacpayne can I get a review?

cormacpayne commented on global.json:
```diff
@@ -27,5 +27,8 @@
     "src/ResourceManagement/Monitor/Microsoft.Azure.Monitor"
-  ]
+  ],
+  "sdk": {
```
@jaredmoo why was this added?

jaredmoo replied:
I have VS2017 installed on my machine, without this the build is totally broken. See https://github.com/Azure/azure-sdk-for-net/issues/2922 .

cormacpayne commented on src/ClientRuntime/global.json:
```diff
@@ -1,5 +1,8 @@
-                "Microsoft.Rest.ClientRuntime.Log4Net", "Microsoft.Rest.ClientRuntime.Tracing.Tests" ]
+                "Microsoft.Rest.ClientRuntime.Log4Net", "Microsoft.Rest.ClientRuntime.Tracing.Tests" ],
+  "sdk": {
+    "version": "1.0.0-preview2-003121"
```
@jaredmoo same comment

jaredmoo replied:
#2922

cormacpayne commented on src/ResourceManagement/SqlManagement/global.json:
```diff
@@ -1,3 +1,6 @@
-  "projects": [ "../../TestFramework", "Microsoft.Azure.Management.Sql", "Sql.Tests" ]
+  "projects": [ "../../TestFramework", "Microsoft.Azure.Management.Sql", "Sql.Tests" ],
+  "sdk": {
+    "version": "1.0.0-preview2-003121"
```
@jaredmoo same comment

jaredmoo replied:
#2922

cormacpayne commented on src/TestFramework/global.json:
```diff
@@ -1,3 +1,6 @@
-  "projects": [ "Microsoft.Rest.ClientRuntime.Azure.TestFramework", "Microsoft.Azure.Test.HttpRecorder" ]
+  "projects": [ "Microsoft.Rest.ClientRuntime.Azure.TestFramework", "Microsoft.Azure.Test.HttpRecorder" ],
+  "sdk": {
+    "version": "1.0.0-preview2-003121"
```
@jaredmoo same comment

jaredmoo replied:
#2922

cormacpayne commented on src/ResourceManagement/SqlManagement/Sql.Tests/ImportExportScenarioTests.cs:
```diff
@@ -127,24 +127,6 @@ public void TestImportExport(bool preexistingDatabase, string testPrefix, string
                     });
                 }
-
-                // List Server import/export operations
```
@jaredmoo is this removed block being tested somewhere else? Making sure there isn't any test regression with this being removed

jaredmoo replied:
This API is removed so no test is needed.

jaredmoo replied:
Fixed in 83501016bb0744278a7d4a388985c2c37fc80258

cormacpayne commented on src/ResourceManagement/SqlManagement/Sql.Tests/BlobAuditingTest.cs:
```diff
@@ -0,0 +1,183 @@
+﻿using Microsoft.Azure.Management.Resources;
```
@jaredmoo this file is missing the license header

jaredmoo replied:
Fixed in 8350101

cormacpayne commented on src/ResourceManagement/SqlManagement/Sql.Tests/ThreatDetectionTest.cs:
```diff
@@ -0,0 +1,166 @@
+﻿using Microsoft.Azure.Management.Resources;
```
@jaredmoo this file is missing license header

jaredmoo replied:
Fixed in 83501016bb0744278a7d4a388985c2c37fc80258

cormacpayne reviewed (changes requested):
@jaredmoo some comments that need to be resolved

Also, what is the `UpgradeLog.htm` file? I was unable to view it on GitHub or CodeFlow

jaredmoo committed:
Added license header

jaredmoo committed:
Removed UpgradeLog.htm

jaredmoo commented:
`UpgradeLog.htm` is visual studio junk. Delete it.

jaredmoo commented:
@cormacpayne - I have addressed feedback. Not changing project.json since then it doesn't build for me. I think project.json goes away when #2815 is done.

cormacpayne commented:
@jaredmoo it looks like there is a merge conflict from a pull request that was merged earlier today

jaredmoo committed:
Merge branch 'AutoRest' into breakingchanges

jaredmoo commented:
Ok, I reverted my change to this file in order to resolve the conflict.

jaredmoo commented:
@cormacpayne ready to merge again :)

jaredmoo committed:
Reverted global.json changes
